### PR TITLE
Renamed whole Mac OS X's to macOS

### PR DESCRIPTION
**Pull request by ergenekonyigit**
I renamed whole different variations Mac OS X's to macOS

A few different variations I found.

Mac OS 		——>	macOS
Mac OS X 	——>	macOS
Mac OSX 	——>	macOS
MacOS X		——>	macOS
Mac OSX		——>	macOS
MacOSX 		——>	macOS
OSX 		——>	macOS
OS X 		——>	macOS

With this, everything about macOS can be found comfortably.

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately our records indicate you have not signed the CLA. For legal reasons we need you to sign this before we can look at your contribution. Please follow these steps to rectify the issue:

1. Sign the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/)
2. **Wait at least a day** and then check "Your Details" on [bugs.python.org](https://bugs.python.org/) to see if your account has been marked as having signed the CLA (the delay is due to a person having to manually check your signed CLA)
3. Reply here saying you have completed the above steps


Thanks again to your contribution and we look forward to looking at it!

File: Doc/library/macpath.rst [hunk 1]
```diff
@@ -1,15 +1,15 @@
-:mod:`macpath` --- Mac OS 9 path manipulation functions
+:mod:`macpath` --- macOS 9 path manipulation functions
```
Comment (methane) [inline Doc/library/macpath.rst]: macOS 9 is OK?

File: Doc/library/select.rst [hunk 2]
```diff
@@ -516,7 +516,7 @@ https://www.freebsd.org/cgi/man.cgi?query=kqueue&sektion=2
    | :const:`KQ_FILTER_PROC`   | Watch for events on a process id            |
    +---------------------------+---------------------------------------------+
    | :const:`KQ_FILTER_NETDEV` | Watch for events on a network device        |
-   |                           | [not available on Mac OS X]                 |
+   |                           | [not available on macOS]                 |
```
Comment (methane) [inline Doc/library/select.rst]: please align right bar

File: Doc/library/webbrowser.rst [hunk 3]
```diff
@@ -136,9 +136,9 @@ for the controller classes, all defined in this module.
 +------------------------+-----------------------------------------+-------+
 | ``'windows-default'``  | :class:`WindowsDefault`                 | \(2)  |
 +------------------------+-----------------------------------------+-------+
-| ``'macosx'``           | :class:`MacOSX('default')`              | \(3)  |
+| ``'macosx'``           | :class:`macOS('default')`              | \(3)  |
 +------------------------+-----------------------------------------+-------+
-| ``'safari'``           | :class:`MacOSX('safari')`               | \(3)  |
+| ``'safari'``           | :class:`macOS('safari')`               | \(3)  |
```
Comment (methane) [inline Doc/library/webbrowser.rst]: alignment

File: Doc/library/macpath.rst (hunk 1 above)
Comment (ergenekonyigit) [reply Doc/library/macpath.rst]: @methane i think it's fine, isn't you?

Comment (methane) [reply Doc/library/macpath.rst]: I don't know what is Apple's definition.  Is there any resource from Apple?

Comment (ergenekonyigit) [reply Doc/library/macpath.rst]: @methane Apple called mac OS 9 but there is no source on their website now.

Comment (methane) [reply Doc/library/macpath.rst]: https://en.wikipedia.org/wiki/MacOS

"OS X" was renamed macOS.  But no information about classic "Mac OS".

Comment (ergenekonyigit) [reply Doc/library/macpath.rst]: @methane okey, I'm changing back old Mac OS's.

**Review by ned-deily — CHANGES_REQUESTED** [review]
@ergenekonyigit Thanks for your suggested changes.  But I think we need to be more careful here.  Many of these changes are incorrect and unnecessary churn.  While it is OK to change generic OS references to macOS, it is not correct to refer to macOS versions prior to 10.12.  See https://en.wikipedia.org/wiki/MacOS for the standard Apple designations.  Also, we should not be changing references in WhatsNew documents for older releases.

File: Doc/library/asyncio-eventloops.rst [hunk 4]
```diff
@@ -115,18 +115,18 @@ configuration. See :ref:`asyncio delayed calls <asyncio-delayed-calls>`.
    :class:`ProactorEventLoop` now supports SSL.
 
 
-Mac OS X
+macOS
 ^^^^^^^^
```
Comment (frewsxcv) [inline Doc/library/asyncio-eventloops.rst]: Looks like the underline should be fixed here.

File: Doc/library/plistlib.rst [hunk 5]
```diff
@@ -1,8 +1,8 @@
-:mod:`plistlib` --- Generate and parse Mac OS X ``.plist`` files
+:mod:`plistlib` --- Generate and parse macOS ``.plist`` files
 ================================================================
```
Comment (frewsxcv) [inline Doc/library/plistlib.rst]: Underline. There's probably a lot more instances.

**Comment by ergenekonyigit** [comment]
@ned-deily https://github.com/apple/swift/blob/30dd1a90474032717332f26aca8acbfda427d8bb/test/stdlib/DispatchDeprecationMacOS.swift#L8 Before changing I'm looking in swift source code. Maybe not today but someday i think we have to change them. Look at 11 different variations, this is very complicated, what do you think?

**Comment by berkerpeksag** [comment]
FWIW, I agree with @ned-deily. I suggest to close this PR.

**Comment by brettcannon** [comment]
I agree w/ @ned-deily and @berkerpeksag about this being unnecessary updates. While we should definitely use the proper term going forward, the docs should just be left as-is. Thanks for the PR regardless, @ergenekonyigit !

**Comment by ned-deily** [comment]
FTR, I'm not opposed to updating references in the current docs as long as they are done carefully and accurately.  As @brettcannon suggests, a better way to do that is to make appropriate updates as we move forward with other, more important changes in the various areas.

**Pull request closed without merging**